### ISSUE TITLE
[dashboard][serve] Improve the icon in the tab

### ISSUE THREAD

**Issue by edoakes** [ray-project/ray#34077]
### Description

This is using the old REST API, presumably it'd be application name. Maybe we should title it "Serve Application - <name>" or "Serve - <name>" ?

![Screen Shot 2023-04-04 at 5 47 08 PM](https://user-images.githubusercontent.com/9871461/229938817-f5671390-ebdc-44d3-a00c-5da86ed048bd.png)

**Comment by scottsun94** [comment]
hmm. Where do you see this? It's a bit hard to follow. In the serve application table?

**Comment by edoakes** [comment]
Oh sorry lol the screenshot isn't as useful as I expected 😆

This is in your browser tab while on the application tab (the `-` is the application name on the dashboard).

**Comment by scottsun94** [comment]
I see. So if users don't name the application, serve uses `-` by default?

**Comment by edoakes** [comment]
I think it's actually empty in the Serve REST API response, so probably @alanwguo made it `-` so it's less confusing

**Comment by alanwguo** [comment]
I'll add this to the polish items list

**Comment by scottsun94** [comment]
I would suggest something like this? "- | Serve Application"
The part of info that differentiates itself from other pages should be put at the beginning.

The reason for that is that there can be a lot of tabs and titles will be truncated at a certain points.
"serve application foo" may just become "Serve app...". It doesn't tell which tab is which if there are three "Serve app..."

**Comment by scottsun94** [comment]
feedback from other members on the UX team: 
- "-" doesn't seem meaningful to users and it feels like something goes wrong on this page.
- Should serve generate a more meaningful serve app name? @edoakes

**Comment by edoakes** [comment]
Yeah maybe we should name it "default" or similar instead... @zcin @sihanwang41 any specific reason for the empty default name?

**Comment by zcin** [comment]
Asking @sihanwang41 to chime in here if there's any specific considerations.

I did have a draft PR with an attempt to change the default name to a non-empty string, it broke a lot of tests so I de-prioritized it relative to the other multi-app tasks.
